creator/ingest_runs/tasks.py

```diff
@@ -1,13 +1,109 @@
-def run_ingest(ingest_run_id):
-    # TODO
-    print(f"Preparing ingest run {ingest_run_id}")
+import django_rq
+import logging
 
+from django.core.exceptions import ValidationError
 
-def cancel_ingest(ingest_run_id):
-    # TODO
-    print(f"Cancelling ingest run {ingest_run_id}")
+from creator.decorators import task
+from creator.files.models import FileType
+from creator.ingest_runs.models import IngestRun
+from creator.utils import stop_job
+
+
+logger = logging.getLogger(__name__)
+
+
+@task("run_ingest")
+def run_ingest(ingest_run_id=None):
+    """
+    Run ingest process for a given set of files.
+    """
+    ingest_run = get_ingest_run(ingest_run_id)
+    logging.info(f"Preparing ingest run {ingest_run.pk} for processing.")
+
+    # Check if any IngestRuns with the same input hash exist. If so, cancel
+    # them and delete the newly created IngestRun, which never gets started.
+    # If not, start a new IngestRun.
+    if cancel_duplicate_ingest_runs(ingest_run):
+        ingest_run.delete()
+    else:
+        # Update run state so we know it's running
+        logging.info(f"Start ingest {ingest_run.id}")
```

Comment:
Since the above log statement uses the `.pk` field, let's be consistent and use that in all of the log statements. And I like `.pk` better than `.id` anyway because if we later change the primary key to something other than `id` then we don't have to update these log statements.